server/index.js

```diff
@@ -1,168 +1,21 @@
-      res.send(JSON.stringify(files));
-    });
-  });
-  server.get('/_api/content/:type', (req, res) => {
-    return content.list(req.params.type, files => {
-      res.send(JSON.stringify(files));
-    });
-  });
-
-  server.get('/_api/content/:type/:slug', (req, res) => {
-    return content
-      .load(req.params.type, req.params.slug, data => {
-        res.send(JSON.stringify(data));
-      })
-      .catch(error => {
-        res.status(500);
-        res.send(JSON.stringify({ error: error.toString(), message: error.stack }));
-      });
-  });
-
-  server.delete('/_api/content/:type/:slug', (req, res) => {
-    return content.deleteContent(req.params.type, req.params.slug, data => {
-      res.send();
-    });
-  });
-
-  server.delete('/_api/content/media', (req, res) => {
-    const path = JSON.parse(req.query.name).url;
-    return content.removeMedia(path, result => {
-      res.send(result);
-    });
-  });
-
-  server.post('/_api/content/media', (req, res) => {
-    const maxFilesCount = 12;
-    const promise = new Promise((resolve, reject) => {
-      upload.array('file', maxFilesCount)(req, res, err => {
-        if (err) {
-          reject(err);
-        }
-
-        resolve();
-      });
-    });
-
-    promise
-      .then(() => {
-        return res.send(true);
-      })
-      .catch(err => {
-        return res.status(400).send(true);
-      });
-  });
-
-  server.post('/_api/content/:type/:slug', (req, res) => {
-    if (req.body['media-data']) {
-      const currentDate = `${new Date().getMonth() +
-        1}.${new Date().getDate()}.${new Date().getFullYear()}`;
-
-      req.body['media-data'].forEach((item, index) => {
-        const uuid = uuidv4();
-        content.saveMedia(item, currentDate, uuid);
-        req.body['media-data'][index] = uuid;
-      });
-    }
-
-    return content.save(
-      req.params.type,
-      req.params.slug,
-      JSON.stringify(req.body, null, 2),
-      data => {
-        res.send(JSON.stringify(data));
-      },
-    );
-  });
-
-  server.post('/_api/content/:type', (req, res) => {
-    return content.create(req.params.type, JSON.stringify(req.body, null, 2), data => {
-      res.send(JSON.stringify(data));
-    });
-  });
-
-  server.get('*', (req, res) => {
-    handle(req, res);
-  });
-
-  server.listen(port, err => {
-    if (err) throw err;
-  });
+const cors = require('cors');
+const path = require('path');
+const contentRouter = require('./contentRouter');
+const contentTypesRouter = require('./contentTypesRouter');
+const gitRouter = require('./gitRouter');
+
+const app = express();
+const port = parseInt(process.env.PORT, 10) || 3020;
+const root = path.resolve(__dirname, '..');
+
+app.use(bodyParser.json()); // support json encoded bodies
+app.use(bodyParser.urlencoded({ extended: true })); // support encoded bodies
+app.use(cors());
+app.use('/content-types', contentTypesRouter(root));
+app.use('/modified-files', gitRouter(root));
```

Comment:
looks like '/modified-files' us redundant here as gitRouter() has `router.get('/modified-files')`

Reply:
This is the same problem as in the content router.

Reply:
i just meant it can be `app.use(gitRouter(root));`, but looks like `app.use('<url_pattren>', middleware)` is also a valid signature